src/context.types.ts

```diff
@@ -44,6 +44,10 @@ export interface Context {
    * The current job run context
    */
   jobRun: JobRunContext & Pick<GithubContext, GithubContextJobRunPropertyKeys>
+  /**
+   * Other data
+   */
+  data: any | undefined
```

Comment:
```suggestion
  data?: any
```

I would prefer that we actually have typed data here. We'll need to know the structure of it anyway once we write loki queries against it.

Reply:
Another Q, do we want to add an additional property to `Context` so we can make faster filters against our logs? We can consider creating context as the result of a [discriminated union](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions). An example of a property to be used as a discriminator could be called something like `testType`, with its possible values being `CHAINLINK_GO_CORE_TESTS | CHAINLINK_SOLIDITY_CORE_TESTS`.

This would make it explicit on what `data` contains based on checking the `testType` property, super useful for writing fast loki queries.